### PR TITLE
Allow changing the listen port for the Zope process.

### PR DESCRIPTION
**Pull request by Rudd-O**

### CODE REVIEW

**Comment by Rudd-O** [comment]
I can't run the tests @stevepiercy

**Review by stevepiercy — APPROVED** [review]
Docs LGTM.

A maintainer should review and approve before merging.

**Comment by stevepiercy** [comment]
The test build fails, and we need a maintainer to fix it. See https://github.com/plone/plone-backend/actions/runs/3170279512/jobs/5162762000

**Comment by Rudd-O** [comment]
We'll havetawait then.

**Comment by Rudd-O** [comment]
So, is the Github action set fixed?  I can gladly fix the conflict as long as the test suite is fixed, or we have an exception to merge it through.

**Comment by Rudd-O** [comment]
Where do I check in the documentation for this change?

```
 +### Listen port
+
+By default, the Zope process inside the container will listen on TCP port 8080.
+In certain circumstances — Kubernetes or Podman pods — there may be a need to run
+more than one Zope process within the network namespace, which would result in
+listen port clashes.
+
+In these cases, the variable `LISTEN_PORT` can be set to any particular port above
+1024 to ensure that the container will listen on the desired port.
+
```

**Review by jensens — APPROVED** [review]
Ok to merge given tests are green.

**Comment by stevepiercy** [comment]
> Where do I check in the documentation for this change?
> 
> ```
>  +### Listen port
> +
> +By default, the Zope process inside the container will listen on TCP port 8080.
> +In certain circumstances — Kubernetes or Podman pods — there may be a need to run
> +more than one Zope process within the network namespace, which would result in
> +listen port clashes.
> +
> +In these cases, the variable `LISTEN_PORT` can be set to any particular port above
> +1024 to ensure that the container will listen on the desired port.
> +
> ```

Unless someone says differently, I would include it in the docs at [Main variables](https://6.dev-docs.plone.org/install/containers/images/backend.html#main-variables). Repo: https://github.com/plone/documentation then request review from me and @jensens

**Comment by Rudd-O** [comment]
@jensens tests are not green:

```
Run docker/login-action@v1
Warning: The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
Warning: The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
Warning: The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
Error: Username and password required
```

**Comment by jensens** [comment]
@ericof do you have a clue whats wrong here, since it used to work?

**Comment by davisagli** [comment]
@jensens @Rudd-O When the github actions run for a PR from a fork, they do not have access to this repository's github secrets for logging in to Dockerhub. Please push your branch to plone/plone-backend instead of Rudd-O/plone-backend and open a new PR.

**Comment by Rudd-O** [comment]
OK, please stand by.

**Comment by Rudd-O** [comment]
I can't push to `plone/plone-backend` either:

```
[user@projects plone-backend]$ git checkout -B listenport
Switched to a new branch 'listenport'
[user@projects plone-backend]$ git push upstream listenport:listenport
ERROR: Permission to plone/plone-backend.git denied to Rudd-O.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
[user@projects plone-backend]$ cat .git/config
...
[remote "upstream"]
	url = git@github.com:plone/plone-backend.git
	fetch = +refs/heads/*:refs/remotes/upstream/*
...
```

Either way, this is not optimal that people can't open pull requests to this repo because the tests are misconfigured.  Please fix the underlying cause.

**Comment by stevepiercy** [comment]
I think the repo settings has not added members the team [`@plone/developers`](https://github.com/orgs/plone/teams/developers) under https://github.com/plone/plone-backend/settings/access.

Here is `plone/documentation` permissions:
Screen Shot 2022-11-09 at 4 41 10 PM

**Comment by davisagli** [comment]
@jensens @ericof The @plone/developers team currently has Triage permission on this repository, which doesn't include permission to push commits to it. Can I change that to the Write permission?

**Review by tiberiuichim — APPROVED** [review]

**Comment by stevepiercy** [comment]
Also in `plone/documentation` we use a protected branch to prevent this group from merging PRs. We have enabled:

- Restrict who can dismiss pull request reviews
- Allow specified actors to bypass required pull requests
- Require status checks to pass before merging
- Restrict who can push to matching branches
  - Organization administrators, repository administrators, and users with the Maintain role.
  - [plone/documentation-team](https://github.com/orgs/plone/teams/documentation-team)

This prevents n00bs merging their own PRs.

File: skeleton/docker-entrypoint.sh
Comment (busykoala) [inline skeleton/docker-entrypoint.sh]: What is the use case of running the same container twice in one pod? Generally, I would argue that it would make more sense to have multiple pods and scale them up/down with a deployment instead of scaling up the containers per pod.
The only reason for this I can come up with is if the proxy overhead is getting too high in a service mesh like istio.

**Comment by jensens** [comment]
I fixed the repo settings.

**Review by jensens — APPROVED** [review]

File: skeleton/docker-entrypoint.sh
Comment (busykoala) [reply skeleton/docker-entrypoint.sh]: Additionally, I would suggest that if this was an option the `EXPOSE` in the Dockerfile be removed if this was merged, it will be confusing otherwise in my opinion.

**Comment by ericof** [comment]
I would like to know what is the use case for this.  I agree with @busykoala that this would require us to:

- Remove the EXPOSE
- Remove HEALTHCHECK

File: skeleton/docker-entrypoint.sh
Comment (Rudd-O) [reply skeleton/docker-entrypoint.sh]: > What is the use case of running the same container twice in one pod?

Your whole deployment in a pod.  One ZEO, multiple backend containers, possibly 1-N frontend containers.  The sort of thing you would do pretty standard with `podman pod` in a docker-compose fashion.  In Kubernetes you would probably compose your pods differently tho (I can't really conceive of how you'd keep all the instances of all the containers tied up to each other currently, since I haven't used K8s in a while).

@ericof Shouldn't the health check run a command against the running Plone server inside the container?  The health check now has access to the port via environment variable, so it should not be hard to modify it to continue working.

**Comment by Rudd-O** [comment]
I've updated the health check and removed EXPOSE.  No word yet on whether removing EXPOSE will cause a problem for single-container users.

**Comment by Rudd-O** [comment]
This is now superseded by test-passing PR https://github.com/plone/plone-backend/pull/91 .

**Pull request closed without merging**
